benchmark/json2msgpack/simdjson_ondemand.h

```diff
@@ -74,10 +74,11 @@ simdjson2msgpack::to_msgpack(const simdjson::padded_string &json,
       write_byte(0xc2 + doc.get_bool());
       break;
     case simdjson::ondemand::json_type::null:
-      write_byte(0xc0);
-      // As a side effect, the is_null call advances the document's iterator
-      // so that the current_location check would work in all cases.
-      doc.is_null();
+      // We check that the value is indeed null
+      // otherwise: an error is thrown.
+      if(doc.is_null()) {
```

Comment:
```c
simdjson_inline bool document::is_null() noexcept {
  return get_root_value_iterator().is_root_null();
}
```

I wasn't sure about this - I think the root level is special cased for document (not value), and this won't throw for doc specifically. i.e. it returns bool and not `simdjson_result<bool>`

Makes sense for the value iteration

Reply:
```c
simdjson_warn_unused simdjson_inline simdjson_result<bool> value_iterator::get_root_bool() noexcept {
  auto max_len = peek_start_length();
  auto json = peek_root_scalar("bool");
  uint8_t tmpbuf[5+1];
  if (!_json_iter->copy_to_buffer(json, max_len, tmpbuf)) { return incorrect_type_error("Not a boolean"); }
  auto result = parse_bool(tmpbuf);
  if(result.error() == SUCCESS) {
    if (!_json_iter->is_single_token()) { return TRAILING_CONTENT; }
    advance_root_scalar("bool");
  }
  return result;
}
simdjson_inline bool value_iterator::is_root_null() noexcept {
  // If there is trailing content, then the document is not null.
  if (!_json_iter->is_single_token()) { return false; }
  auto max_len = peek_start_length();
  auto json = peek_root_scalar("null");
  bool result = (max_len >= 4 && !atomparsing::str4ncmp(json, "null") &&
         (max_len == 4 || jsoncharutils::is_structural_or_whitespace(json[5])));
  if(result) { advance_root_scalar("null"); }
  return result;
}
```

It seems like use cases like this might benefit from a helper such as `doc.throw_if_not_null()` or changing doc.is_null to simdjson_result